src/NGen.cpp

```diff
@@ -764,7 +823,42 @@ int main(int argc, char* argv[]) {
 
             int delta_time = manager->Simulation_Time_Object->get_output_interval_seconds();
 
-            router->route(number_of_timesteps, delta_time);
+            std::string t_route_config_file_with_path =
+                manager->get_t_route_config_file_with_path();
+            // model for routing
+            models::bmi::Bmi_Py_Adapter py_troute("T-Route", t_route_config_file_with_path, "troute_nwm_bmi.troute_bmi.BmiTroute", true);
+
+            // tell BMI to resize nexus containers
+            int64_t nexus_count = nexus_indexes.size();
+            py_troute.SetValue("land_surface_water_source__volume_flow_rate__count", &nexus_count);
+            py_troute.SetValue("land_surface_water_source__id__count", &nexus_count);
+            // set up nexus id indexes
+            std::vector<int> nexus_df_index(nexus_count);
+            for (const auto& key_value : nexus_indexes) {
+                int id_index = key_value.second;
+
+                // Convert string ID into numbers for T-route index
+                int id_as_int = -1;
+                int sep_index = key_value.first.find(hy_features::identifiers::seperator);
```

Comment:
```suggestion
                size_t sep_index = key_value.first.find(hy_features::identifiers::seperator);
```
To match the return type from `std::string::find`, and the type of `std::string::npos`